### ISSUE TITLE
Using the Travis CI cache

### ISSUE THREAD

**Issue by stephengroat** [pyside/PySide#149]
I'm trying to build a project that uses pyside (chipwhisperer) on Travis and am having issues using both Travis and the Travis cache.

The Travis build time is long (~30 minutes). Using pip, no console output from the build process shows (which can lead to travis timeouts). Are there any known ways of getting console output to keep travis from timing out?

Also, Travis is building in /tmp for all of its build outputs. Is there any way to get this into the .cache/pip folder or an arbitrary folder? That way, travis could cache the build outputs

**Comment by techtonik** [comment]
No output? Can you provide a link?

I long forgot how Travis cache works. What exactly do you think can be cached and how much build time will it save?

**Comment by stephengroat** [comment]
Sure. let me clarify "output" too, I think I might have been confusing earlier. By "output", i mean console output.

Here's with output (https://travis-ci.org/stephengroat/chipwhisperer/builds/160212969). pyside is installed using setuptools/pip, starting on line #967

Here's another example, with pyside installed via pip install pyside (https://travis-ci.org/stephengroat/chipwhisperer/builds/159982793). As you can see, it timeouts. I've been using travis_wait 30 to overcome this, but I was wondering if there's a pip options to see the console.

Also, for caching, I was hoping to be able to redirect the cmake build that's currently going to (/tmp/pip-build-aQiU5m/pyside/pyside_package). By caching that, i was hoping to be able to take down the build time somwhat

**Comment by techtonik** [comment]
So the issue is that command `pip install pyside` timeouts and doesn't produce any output? Do you have an example without that `travis_wait` to see what actually happens?

**Comment by techtonik** [comment]
And `pip` does have verbose output - try it:
```
  -v, --verbose               Give more output. Option is additive, and can be
                              used up to 3 times.
```

**Comment by stephengroat** [comment]
Here's a `-vvv` (https://travis-ci.org/stephengroat/chipwhisperer/builds/160906100)

Here's a `travis_wait 30`. All it's doing is allowing no console output to work (https://travis-ci.org/stephengroat/chipwhisperer/builds/160907551)

any idea on moving the output that's going to the `/tmp` and getting that `pip-build-[rand]` directory out of there and moving it to a repeatable place?

**Comment by techtonik** [comment]
Before anything else, upgrade `pip` to latest. `6.0.7` is too old to mess with.

**Comment by stephengroat** [comment]
That's the `pip` version included with the default travis python 2.7 environment. Do you really think it's that? I'd rather not have to try to figure out how to update `pip`.

It obviously builds, I'm just wondering if 1. you can redirect build files (such as the `.o`s from the Qt build) and 2. can you redirect the `stdout` from `pip`

**Comment by techtonik** [comment]
Travis CI got some updates, like an ability to run Docker containers. It is probably worth to close this issue and reopen a new one with fresh updates if it is still actual.
